### PR TITLE
Core: reserve fee as an .env variable

### PR DESCRIPTION
**Pull request by talvasconcelos**
Have the fee reserve as an env variable, defaults to 4000 msats

### CODE REVIEW

**Review by callebtc — CHANGES_REQUESTED** [review]
Great PR! Just some unnecessary changes to clean up, otherwise mergable.

File: .env.example [hunk 1]
```diff
@@ -27,6 +27,7 @@ LNBITS_DATA_FOLDER="./data"
 
 LNBITS_FORCE_HTTPS=true
 LNBITS_SERVICE_FEE="0.0"
+LNBITS_FEE_RESERVE=4000 #value in millisats
```
Comment (callebtc) [inline .env.example:30]:
```suggestion
LNBITS_FEE_RESERVE_MIN=4000 # value in millisats
LNBITS_FEE_RESERVE_PERCENT=1 # value in percent
```

File: lnbits/core/services.py [hunk 2]
```diff
@@ -340,4 +340,4 @@ async def check_invoice_status(
 
 # WARN: this same value must be used for balance check and passed to WALLET.pay_invoice(), it may cause a vulnerability if the values differ
 def fee_reserve(amount_msat: int) -> int:
-    return max(2000, int(amount_msat * 0.01))
+    return max(int(RESERVE_FEE), int(amount_msat * 0.01))
```
Comment (callebtc) [inline lnbits/core/services.py:343]:
```suggestion
    return max(int(RESERVE_FEE), int(amount_msat * LNBITS_FEE_RESERVE_MIN/100.0))
```

File: lnbits/settings.py [hunk 3]
```diff
@@ -52,6 +51,7 @@
 DEFAULT_WALLET_NAME = env.str("LNBITS_DEFAULT_WALLET_NAME", default="LNbits wallet")
 PREFER_SECURE_URLS = env.bool("LNBITS_FORCE_HTTPS", default=True)
 
+RESERVE_FEE = env.int("LNBITS_RESERVE_FEE", default=4000)
```
Comment (callebtc) [inline lnbits/settings.py:54]:
```suggestion
RESERVE_FEE_MIN = env.int("LNBITS_RESERVE_FEE", default=4000)
RESERVE_FEE_PERCENT = env.int("LNBITS_FEE_RESERVE_PERCENT", default=1)
```

**Comment by callebtc** [comment]
One nitpick: python variable is called `RESERVE_FEE_MIN` and environment variable is `LNBITS_FEE_RESERVE_MIN`

Consistent name like RESERVE_FEE or FEE_RESERVE would be nicer.

File: lnbits/settings.py [hunk 4]
```diff
@@ -52,6 +51,8 @@
 DEFAULT_WALLET_NAME = env.str("LNBITS_DEFAULT_WALLET_NAME", default="LNbits wallet")
 PREFER_SECURE_URLS = env.bool("LNBITS_FORCE_HTTPS", default=True)
 
+RESERVE_FEE_MIN = env.int("LNBITS_RESERVE_FEE_MIN", default=4000)
+RESERVE_FEE_PERCENT = env.int("LNBITS_RESERVE_FEE_PERCENT", default=1)
```
Comment (callebtc) [inline lnbits/settings.py:55]:
```suggestion
RESERVE_FEE_PERCENT = env.float("LNBITS_RESERVE_FEE_PERCENT", default=1)
```

File: .env.example [hunk 5]
```diff
@@ -27,6 +27,8 @@ LNBITS_DATA_FOLDER="./data"
 
 LNBITS_FORCE_HTTPS=true
 LNBITS_SERVICE_FEE="0.0"
+LNBITS_RESERVE_FEE_MIN=4000 # value in millisats
+LNBITS_RESERVE_FEE_PERCENT=1 # value in percent
```
Comment (callebtc) [inline .env.example:31]:
```suggestion
LNBITS_RESERVE_FEE_PERCENT=1.0 # value in percent
```

**Comment by callebtc** [comment]
@talvasconcelos, unfortunately tests are still failing (bleskomat extension?).

**Comment by callebtc** [comment]
Let's get this PR back out of the grave. Any reason why the default is increased from 2 sat to 4 sat?

**Comment by talvasconcelos** [comment]
> Let's get this PR back out of the grave. Any reason why the default is increased from 2 sat to 4 sat?

i think it was Ben's call!

File: tests/extensions/bleskomat/test_lnurl_api.py [hunk 6]
```diff
@@ -133,4 +133,4 @@ async def test_bleskomat_lnurl_api_action_success(client, lnurl):
     assert wallet.balance_msat == 50000
     bleskomat_lnurl = await get_bleskomat_lnurl(secret)
     assert bleskomat_lnurl.has_uses_remaining() == False
-    WALLET.pay_invoice.assert_called_once_with(pr, 2000)
+    WALLET.pay_invoice.assert_called_once_with(pr, 4000)
```
Comment (callebtc) [inline tests/extensions/bleskomat/test_lnurl_api.py:136]:
```suggestion
    WALLET.pay_invoice.assert_called_once_with(pr, 2000)
```

File: .env.example [hunk 7]
```diff
@@ -25,6 +25,8 @@ LNBITS_DATA_FOLDER="./data"
 
 LNBITS_FORCE_HTTPS=true
 LNBITS_SERVICE_FEE="0.0"
+LNBITS_RESERVE_FEE_MIN=4000 # value in millisats
```
Comment (callebtc) [inline .env.example:28]:
```suggestion
LNBITS_RESERVE_FEE_MIN=2000 # value in millisats
```

File: lnbits/settings.py [hunk 8]
```diff
@@ -55,6 +55,8 @@
 DEFAULT_WALLET_NAME = env.str("LNBITS_DEFAULT_WALLET_NAME", default="LNbits wallet")
 PREFER_SECURE_URLS = env.bool("LNBITS_FORCE_HTTPS", default=True)
 
+RESERVE_FEE_MIN = env.int("LNBITS_RESERVE_FEE_MIN", default=4000)
```
Comment (callebtc) [inline lnbits/settings.py:58]:
```suggestion
RESERVE_FEE_MIN = env.int("LNBITS_RESERVE_FEE_MIN", default=2000)
```

**Pull request merged**